azure-pipelines.yml

```diff
@@ -40,8 +40,16 @@ stages:
         displayName: "Install NuGet"
 
       - script: nuget restore -LockedMode -ConfigFile "analyzers\NuGet.Config" $(solution)
+        displayName: "NuGet Restore"
 
-      - powershell: .\scripts\build\set-azp-project-version-variable.ps1
+      - powershell: .\scripts\build\store-azp-variables.ps1
+        displayName: "Store AZP Variables"
+
+      - publish: $(Agent.BuildDirectory)/Azp-Variables
+        artifact: Azp-Variables
+        displayName: "Publish AZP Variables as pipeline artifact"
+
+      - template: set-azp-variables-steps.yml@commonTemplates
```

Comment:
This downloads the artifact that was just published on the previous line, but it takes only 1s so I'm keeping it as is for simplicity/consistency.